src/LVM.cpp

```diff
@@ -897,38 +903,38 @@ void LVM::execute(std::unique_ptr<LVMCode>& codeStream, InterpreterContext& ctxt
                 break;
             }
             case LVM_Fact: {
-                std::string relName = symbolTable.resolve(code[ip + 1]);
+                size_t relId = code[ip + 1];
                 auto arity = code[ip + 2];
                 RamDomain tuple[arity];
                 for (auto i = 0; i < arity; ++i) {
                     tuple[arity - i - 1] = stack.top();
                     stack.pop();
                 }
-                getRelation(relName).insert(tuple);
+                getRelation(relId)->insert(tuple);
                 ip += 3;
                 break;
             }
             case LVM_Merge: {
-                std::string source = symbolTable.resolve(code[ip + 1]);
-                std::string target = symbolTable.resolve(code[ip + 2]);
+                size_t sourceId = code[ip + 1];
+                size_t targetId = code[ip + 2];
                 // get involved relation
-                InterpreterRelation& src = getRelation(source);
-                InterpreterRelation& trg = getRelation(target);
+                auto srcPtr = getRelation(sourceId);
+                auto trgPtr = getRelation(targetId);
 
-                if (dynamic_cast<InterpreterEqRelation*>(&trg)) {
+                if (dynamic_cast<InterpreterEqRelation*>(trgPtr)) {
```

Comment:
Explicitly compare to nullptr
```suggestion
                if (dynamic_cast<InterpreterEqRelation*>(trgPtr) != nullptr) {
```